### PR TITLE
[TIP] Use Kibana ui settings for date

### PR DESCRIPTION
**Pull request by PhilippeOberti**
## Summary

This PR uses Kibana uiSettings to retrieve user's preferred date format and timezone. The change has been applied to the Flyout subtitle (see in the video below)

I added a new `DateFormatter` component (to leverage the new `useDateFormat` and `useTimeZone` custom hooks). The component also uses the prior dateFormatter utils function.

### Checklist

- [ ] create Storybook entry for the new `DateFormatter` component
- [ ] add unit tests to the new component and update unit tests for everything else

Here's a video to show how changing the dateFormat property in the `Advanced Settings` section of Kibana changes the display in the TI plugin

https://user-images.githubusercontent.com/17276605/179250651-caaea7de-36aa-45bb-8016-891866e8aa05.mov

https://github.com/elastic/security-team/issues/4369

### CODE REVIEW

File: x-pack/plugins/threat_intelligence/public/common/mocks/test_providers.tsx [hunk 1]
```diff
@@ -8,11 +8,29 @@
...
+  }
+}
+
+export const TestProvidersComponent: VFC<Props> = ({ children }) => {
+  const mockCoreStart = coreMock.createStart();
+  mockCoreStart.uiSettings.get.mockImplementation(uiSetting);
+
+  return (
+    <I18nProvider>
+      <KibanaContextProvider services={mockCoreStart}>{children}</KibanaContextProvider>
```
Comment (PhilippeOberti) [inline x-pack/plugins/threat_intelligence/public/common/mocks/test_providers.tsx:33]: added this Provider to be able to get uiSettings in our jest tests

File: x-pack/plugins/threat_intelligence/public/common/utils/dates.tsx [hunk 2]
```diff
@@ -9,9 +9,12 @@ import moment from 'moment';
...
-
-export const fullDateFormatter = (date: string | moment.Moment): string => {
-  const momentDate: moment.Moment = typeof date === 'string' ? moment(date) : date;
-  return momentDate.isValid() ? momentDate.format(FULL_DATE) : EMPTY_VALUE;
+export const dateFormatter = (
+  date: string | moment.Moment,
+  timeZone: string,
+  format?: string
+): string => {
+  const momentDate: moment.Moment =
```
Comment (PhilippeOberti) [inline x-pack/plugins/threat_intelligence/public/common/utils/dates.tsx:17]: the formatter now handles timezones automatically

**Comment by lgmys** [comment]
Hey, I think you need to cherry pick that commit on top of a new branch, created from main - since you altered the branch history :)

**Comment by PhilippeOberti** [comment]
> Hey, I think you need to cherry pick that commit on top of a new branch, created from main - since you altered the branch history :)

Yeah I as thinking about doing this only after we merge [the PR to upstream](https://github.com/elastic/kibana/pull/136479) to avoid having to do it too many times. Unless we want to get this in before? It didn't seem crucial to me, that's why I didn't worry about it

**Review by lgmys — APPROVED** [review]

**Comment by PhilippeOberti** [comment]
Closing this PR as I opened [a new one](https://github.com/elastic/kibana/pull/137155), from my fork and directly to elastic/kibana repo's main branch

**Pull request closed without merging**